slackbot.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2018 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"time"
+
+	log "github.com/gravitational/logrus"
+	"github.com/gravitational/trace"
+
+	pagerduty "github.com/PagerDuty/go-pagerduty"
+	"github.com/shomali11/slacker"
+)
+
+// Init is called upon Bot creation (first startup)
+func Init(config *Config) {
+	log.Println("Connected!")
+	client := pagerduty.NewClient(config.PagerDuty.APIKey)
+	var opts pagerduty.GetScheduleOptions
+	if schedule, err := client.GetSchedule(config.PagerDuty.Schedule, opts); err != nil {
+		trace.Wrap(err)
+	} else {
+		log.Println("Configured schedule is \"" + schedule.Name + "\" with ID: " + config.PagerDuty.Schedule)
+	}
+}
+
+// Err function is used to handle all Error reported by the Bot
+func Err(err string) {
+	fmt.Println(err)
+}
+
+// Emergency is used to open Emergency Incidents on PagerDuty
+func Emergency(request slacker.Request, response slacker.ResponseWriter, config *Config) {
+	client := pagerduty.NewClient(config.PagerDuty.APIKey)
+	var scheduleOpts pagerduty.GetScheduleOptions
+	if schedule, err := client.GetSchedule(config.PagerDuty.Schedule, scheduleOpts); err != nil {
+		trace.Wrap(err)
+	} else {
```

Comment:
The same for else here

Reply:
Also as I remember schedule variable is available only inside of if scope. You have to use
```
schedule, err := client.GetSchedule(config.PagerDuty.Schedule, scheduleOpts)
if err != nil {
   ...
}
```

Reply:
no I cannot remove the `else` (as noted elsewhere) cause `schedule` as you correctly pointed out is available only inside the `if` statement scope.
I would have to create a different variable available on the outer scope to remove the else which would be worse (?) than the else statement... I guess :thinking:

Reply:
You create `schedule` variable anyway, does not matter inside if or outside. Just use typical snippet:
```
var, err := function()
if err != nil {
  do something
}
```

Reply:
Fixed.